enderio-decoration/src/main/java/com/enderio/decoration/common/blockentity/PoweredLightBlockEntity.java

```diff
@@ -0,0 +1,208 @@
+					}
+				}
+				boolean bl = false;
+				for (BlockPos blockedpos: blocked) {
+					if (isBlocked(relative, center, blockedpos) && level.getBlockState(relative).is(DecorBlocks.LIGHT_NODE.get())) {
+						level.setBlock(relative, Blocks.AIR.defaultBlockState(), 1);
+						bl = true;
+					}
+				}
+				if (!level.getBlockState(relative).isAir() && !level.getBlockState(relative).is(DecorBlocks.LIGHT_NODE.get()) && !bl) {
+					blocked.add(relative);
+				}
+			}
+		}
+		return next;
+	}
+	
+	/**
+	 * Check if the light is blocked by another block. For now this means any that the block is inside a cube 
+	 * created away from the center, with the blocked position as edge.
+	 * @param node
+	 * @param center
+	 * @param blocked
+	 * @return
+	 */
+	private static boolean isBlocked(BlockPos node, BlockPos center, BlockPos blocked) {
+		boolean x = false;
+		boolean y = false;
+		boolean z = false;
+		if (center.getX() >= blocked.getX() && blocked.getX() >= node.getX()) {
+			x = true;
+		}
+		if (center.getX() <= blocked.getX() && blocked.getX() <= node.getX()) {
+			x = true;
+		}
+		if (center.getY() >= blocked.getY() && blocked.getY() >= node.getY()) {
+			y = true;
+		}
+		if (center.getY() <= blocked.getY() && blocked.getY() <= node.getY()) {
+			y = true;
+		}
+		if (center.getZ() >= blocked.getZ() && blocked.getZ() >= node.getZ()) {
+			z = true;
+		}
+		if (center.getZ() <= blocked.getZ() && blocked.getZ() <= node.getZ()) {
+			z = true;
+		}
+		return x && y && z;
+	}
+	
+	/**
+	 * Returns true if the block is inside the zone defined by the spread.
+	 * @param node
+	 * @param center
+	 * @return
+	 */
+	public static boolean inSpreadZone(BlockPos node, BlockPos center) { 
+		if (Math.abs(node.getX()-center.getX()) > spread) {
+			return false;
+		}
+		if (Math.abs(node.getY()-center.getY()) > spread) {
+			return false;
+		}
+		if (Math.abs(node.getZ()-center.getZ()) > spread) {
+			return false;
+		}
+		return true;
+	}
+	
+	private static void consumePower(Level level, BlockPos pos, BlockState state, PoweredLightBlockEntity e) {
+		BlockEntity be = level.getBlockEntity(pos.relative(state.getValue(Light.FACING).getOpposite()));
+		if (be != null) {
+			LazyOptional<IEnergyStorage> energy = be.getCapability(CapabilityEnergy.ENERGY, state.getValue(Light.FACING));
+			if (energy.isPresent()) {
+				if (energy.resolve().get().extractEnergy(RF_USE_TICK, true) == RF_USE_TICK) {
+					boolean powered = level.hasNeighborSignal(pos);
+					if (powered == ((Light) state.getBlock()).isInverted() ? state.getValue(Light.ENABLED) : !state.getValue(Light.ENABLED)) {
+					    e.active = false;
+						return;
+					}
+			        level.setBlock(pos, state.setValue(Light.ENABLED, false), 3);
+					e.active = true;
+					energy.resolve().get().extractEnergy(RF_USE_TICK, false);
+					return;
+				}
+			}
+		}
+	}
+	
+	private static void consumePowerWireless(Level level, BlockPos pos, BlockState state, PoweredLightBlockEntity e) {
```

Comment:
If I'm reading this right, this doesn't consume any power?

Reply:
Yes, since we do not have the wireless version. I should have put a comment there. This is just the behavior needed to set the light on.

Reply:
Alrighty, sounds good :) Might be worth writing some power consumption utilities in base/core when we do implement wireless so we can reuse the capability elsewhere later.

Reply:
Still needs a todo here :)